cmd/transportRequestUploadRFC_test.go

```diff
@@ -50,61 +52,90 @@ func (m *uploadMock) WithConnection(c rfc.Connection) {
-
-	mock := uploadMock{}
+type configMock struct {
+	config *transportRequestUploadRFCOptions
+}
 
-	err := runTransportRequestUploadRFC(&config, &mock, nil, utils)
+func TestTrRfcRunTransportRequestUpload(t *testing.T) {
+	t.Parallel()
 
-	if assert.NoError(t, err) {
-		t.Run("upload triggered", func(t *testing.T) {
-			assert.True(t, mock.uploadCalled)
-		})
-		t.Run("parameters has been marshalled", func(t *testing.T) {
-			assert.Equal(t, rfc.UploadAction{
-				Connection: rfc.Connection{
-					Endpoint: "https://my.abap.server",
-					Client:   "001",
-					Instance: "00",
-					User:     "me",
-					Password: "******",
-				},
-				Application: rfc.Application{
-					Name:        "MyApp",
-					Description: "Lorem impsum",
-					AbapPackage: "XX",
-				},
-				Configuration: rfc.UploadConfig{
-					AcceptUnixStyleEndOfLine: true,
-					CodePage:                 "UTF-8",
-					FailUploadOnWarning:      true,
-					Verbose:                  false, // comes from general config
-				},
-				TransportRequestID: "XXXK12345678",
-				ApplicationURL:     "http://example.org/myDeployable.zip",
-			}, mock.received)
+	t.Run("good", func(t *testing.T) {
+		t.Parallel()
+
+		utils := newTransportRequestUploadRFCTestsUtils()
+		configMock := newRfcConfigMock()
+		actionMock := uploadMock{}
+		cpe := &transportRequestUploadRFCCommonPipelineEnvironment{}
+
+		err := runTransportRequestUploadRFC(configMock.config, &actionMock, nil, utils, cpe)
+
+		if assert.NoError(t, err) {
+			t.Run("upload triggered", func(t *testing.T) {
+				assert.True(t, actionMock.uploadCalled)
+			})
+			t.Run("parameters has been marshalled", func(t *testing.T) {
+				assert.Equal(t, rfc.UploadAction{
+					Connection: rfc.Connection{
+						Endpoint: "https://my.abap.server",
+						Client:   "001",
+						Instance: "00",
+						User:     "me",
+						Password: "******",
+					},
+					Application: rfc.Application{
+						Name:        "MyApp",
+						Description: "Lorem impsum",
+						AbapPackage: "ABC",
+					},
+					Configuration: rfc.UploadConfig{
+						AcceptUnixStyleEndOfLine: true,
+						CodePage:                 "UTF-8",
+						FailUploadOnWarning:      true,
+						Verbose:                  false, // comes from general config
+					},
+					TransportRequestID: "K12345678",
+					ApplicationURL:     "http://example.org/myDeployable.zip",
+				}, actionMock.received)
+
+				assert.Equal(t, cpe.custom.transportRequestID, "K12345678")
+			})
+		}
+	})
+	t.Run("bad", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("Error during deployment", func(t *testing.T) {
+			utilsMock := newTransportRequestUploadSOLMANTestsUtils(0)
+			configMock := newRfcConfigMock()
+			actionMock := uploadMock{failWith: fmt.Errorf("upload failed")}
+			cpe := &transportRequestUploadRFCCommonPipelineEnvironment{}
+
+			err := runTransportRequestUploadRFC(configMock.config, &actionMock, nil, utilsMock, cpe)
+
+			assert.Error(t, err, "upload failed")
```

Comment:
I'm running this code through the debugger now, and it doesn't really show that the code fails, does it? It seems more like it checks the variable you just set. Maybe we don't need this test then?

Reply:
indeed, this is a very small test. it makes sure that if the action throws/returns an error, the error is passed to the outside by runTransportRequestUploadRFC.

Reply:
Okay I understand